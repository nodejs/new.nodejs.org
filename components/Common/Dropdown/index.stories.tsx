import Dropdown from './index';
import type { Meta as MetaObj, StoryObj } from '@storybook/react';

type Story = StoryObj<typeof Dropdown>;
type Meta = MetaObj<typeof Dropdown>;

const items = [...Array(10).keys()].map(item => ({
  title: `Item ${item + 1}`,
  label: `item-${item + 1}`,
  active: false,
  onClick: () => {},
}));

items[2].active = true;

export const Default: Story = {
  args: {
    items: items,
    shouldShow: true,
  },
};

export default { component: Dropdown } as Meta;
