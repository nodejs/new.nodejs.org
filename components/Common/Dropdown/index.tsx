import styles from './index.module.scss';
import type { DropdownItem } from '../../../types';
import type { CSSProperties, FC, KeyboardEvent } from 'react';

type DropdownProps = {
  items: Array<DropdownItem>;
  shouldShow: boolean;
  styles: CSSProperties;
};

const Dropdown: FC<DropdownProps> = ({
  items,
  shouldShow,
  styles: extraStyles,
}) => {
  const mappedElements = items.map(item => {
    const extraStyles = { fontWeight: item.active ? 'bold' : 'normal' };

    const handleKeyPress = (e: KeyboardEvent<HTMLButtonElement>) => {
      if (e.key === 'Enter' || e.key === ' ') {
        item.onClick();
      }
    };

    return (
      <li key={`dropdown-item-${item.label}`}>
        <button
          style={extraStyles}
          onClick={item.onClick}
          onKeyDown={handleKeyPress}
          type="button"
        >
          {item.title}
        </button>
      </li>
    );
  });

  const dropdownStyles = {
    display: shouldShow ? 'block' : 'none',
    ...extraStyles,
  };

  return (
    <ul className={styles.dropdownList} style={dropdownStyles}>
      {mappedElements}
    </ul>
  );
};

export default Dropdown;
