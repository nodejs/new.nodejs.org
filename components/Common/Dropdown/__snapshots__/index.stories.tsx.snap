// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Common/Dropdown Default smoke-test 1`] = `
<ul class="Dropdown_dropdownList__hrw_7"
    style="display: block;"
>
  <li>
    <button type="button"
            style="font-weight: normal;"
    >
      Item 1
    </button>
  </li>
  <li>
    <button type="button"
            style="font-weight: normal;"
    >
      Item 2
    </button>
  </li>
  <li>
    <button type="button"
            style="font-weight: bold;"
    >
      Item 3
    </button>
  </li>
  <li>
    <button type="button"
            style="font-weight: normal;"
    >
      Item 4
    </button>
  </li>
  <li>
    <button type="button"
            style="font-weight: normal;"
    >
      Item 5
    </button>
  </li>
  <li>
    <button type="button"
            style="font-weight: normal;"
    >
      Item 6
    </button>
  </li>
  <li>
    <button type="button"
            style="font-weight: normal;"
    >
      Item 7
    </button>
  </li>
  <li>
    <button type="button"
            style="font-weight: normal;"
    >
      Item 8
    </button>
  </li>
  <li>
    <button type="button"
            style="font-weight: normal;"
    >
      Item 9
    </button>
  </li>
  <li>
    <button type="button"
            style="font-weight: normal;"
    >
      Item 10
    </button>
  </li>
</ul>
`;
